### PR TITLE
2404: Convert e2e-tests to esm

### PR DESCRIPTION
**Pull request by sarahsporck**
### Short description

Convert e2e-test project to esm

### Proposed changes

- adding file extensions
- adding options module and moduleResolution to tsconfig.

### Resolved issues

Relates: #2404 

---

### CODE REVIEW

File: e2e-tests/native/wdio-ios.conf.ts [hunk 1]
```diff
@@ -1,9 +1,8 @@
-import { Capabilities } from '@wdio/types/build/Capabilities'
-import { Testrunner } from '@wdio/types/build/Options'
+import { Capabilities } from '@wdio/types'
```
Comment (sarahsporck) [inline e2e-tests/native/wdio-ios.conf.ts:1]: With esm you are only allowed to use 'publicly exported' dependencies

File: web/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "start:malte": "yarn run webpack-dev-server-ts --env config_name=malte --env dev_server",
     "start:malte-test-cms": "yarn run webpack-dev-server-ts --env config_name=malte-test-cms --env dev_server",
     "start:aschaffenburg": "yarn run webpack-dev-server-ts --env config_name=aschaffenburg --env dev_server",
-    "start:integreat-e2e": "yarn run webpack-dev-server-ts --env config_name=integreat-e2e",
+    "start:integreat-e2e": "yarn run webpack-dev-server-ts --env config_name=integreat-e2e --env dev_server",
```
Comment (sarahsporck) [inline web/package.json:12]: I added this, as the bundle analyzer plugin is quite verbose and actually not required here.

Comment (ztefanie) [reply web/package.json:12]: Not sure if this is really a good decision. With e2e tests we want to imitate the real productive app as much as possible, and with settings the dev_server to true, we change quite a lot of the behavior. Maybe we can introduce a new variable for this?

**Review by ztefanie — APPROVED** [review]
Nice PR :tada: 

Thanks for bringing this topic up :+1:

File: web/tools/webpack.config.ts [hunk 3]
```diff
@@ -207,7 +214,7 @@ const createConfig = (
     // The list of plugins for Webpack compiler
     plugins: [
       new BundleAnalyzerPlugin({
-        analyzerMode: devServer ? 'disabled' : 'static',
+        analyzerMode: !bundleAnalyer ? 'disabled' : 'static',
```
Comment (steffenkleinle) [inline web/tools/webpack.config.ts:217]:
```suggestion
        analyzerMode: bundleAnalyer ? 'static' : 'disabled',
```

File: e2e-tests/web/test/specs/navigateToSearchResults.e2e.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
-import { URL } from 'url'
+import { URL } from 'node:url'
 
-import { contentSearch, Routes } from '../../../shared/constants'
-import DashboardPage from '../pageobjects/dashboard.page'
-import SearchPage from '../pageobjects/search.page'
+import { contentSearch, Routes } from '../../../shared/constants.js'
```
Comment (steffenkleinle) [inline e2e-tests/web/test/specs/navigateToSearchResults.e2e.ts:3]: So the extensions are always `js` even for `ts`/`tsx` files?

**Review by steffenkleinle — APPROVED** [review]

**Pull request merged**